### ISSUE TITLE
Initialize() method is called at wrong timing.

### ISSUE THREAD

**Issue by Double-Dude** [MvvmCross/MvvmCross#4005]
## 🐛 Bug Report

Documentation says "Initialize is called after the native navigation has take place.". 
Initialize() method should be invoked after native navigation has taken place, but it's not behaving like this.

The Actual Behaviour is:
Initialize() is called before the Navigation takes place. The method is called immediately after Prepare() method is called. But MvxNavigationService does await the Task after Navigation takes place. But again, it means that the task is actually started immediately after ViewModel is created, Prepare() is called. 

This creates a big issue. Because if you put "Show Loading Spinner" inside the ToBeShowed ViewModel's Initialize() method, the Spinner gets showed before the corresponding ViewController is presented, then when the corresponding ViewController is presented, it actually covered the Loading Spinner.

I've spent a long time digging into the source code and found out the reason. 

![image](https://user-images.githubusercontent.com/12950464/101120856-8bc41100-3653-11eb-96d6-4694219fb74d.png)
![image](https://user-images.githubusercontent.com/12950464/101120899-a4342b80-3653-11eb-9102-18282d7d1470.png)

### Expected behavior
Initialize() method should be invoked after native navigation has taken place, but it's not behaving like this.
### Reproduction steps
1. call _navigationService.Navigate<AnyViewModel>(presentationBundle);
2. ViewModel Initialize() will be called.
3. Then iOS MvxViewPresenter ShowView method is called after ViewModel Initialize() is invoked. The expected behaviour should be the other way round, I.e. Step3 should be called before step 2.
### Configuration

**Version:** 7.1.2

**Platform:** 
- [ ] :iphone: iOS
- [ ] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

**Comment by Double-Dude** [comment]
I actually found this. https://github.com/MvvmCross/MvvmCross/pull/1971

The change was made on intentional, it seems to be. Maybe the documentation needs to be fixed.

**Comment by Hackmodford** [comment]
I've found that there's a lot of documentation that should be updated...

**Comment by Cheesebaron** [comment]
You are right. _You_ can help with that. We can also have the alternative of just removing it. This way no one can complain that the docs are wrong.

**Comment by Hackmodford** [comment]
@Cheesebaron You are even more right. I've been seriously considering it too 😉 
At least with the parts that I understand. There's still a lot about this framework that is a mystery to me. 😆
